### PR TITLE
Added a check when searching for position of header, that the header …

### PR DESCRIPTION
**Pull request by johngray1965**
…found is visible.

Fairly simple change, but it adds another method to StickyRecyclerHeadersAdapter that adapter will need to implement.

### CODE REVIEW

**Comment by jacobtabak** [comment]
Thanks for the PR - can't really introduce a breaking change for an edge case like this, though. But if someone else encounters this problem hopefully they'll find your PR and borrow your code.

**Comment by johngray1965** [comment]
I was toying with the idea of making StickyRecyclerHeadersAdapter an abstract class instead of an interface it default the return value to true.  But then I realized that won't really work out since Java doesn't support multiple inheritance.  So I don't see any easy way to handle this without changing the interface.  If you see a way to handle it better, that would be good.

**Comment by koocbor** [comment]
@johngray1965  - thanks for this.  I encountered this error when a header other than the first one is at the top of recyclerview the first header in the list is returned from findHeaderPositionUnder instead of the one that is actually at the top of the view.  Your PR saved me :+1:

**Comment by johngray1965** [comment]
I'm now consider just adding an independent listener with a setter for it.  That way it won't cause an issue with the rest of the existing code.

**Comment by johngray1965** [comment]
I refactored the change to be backwards compatible.  Any chance it can be merged now?

**Comment by jacobtabak** [comment]
Can you remind me what the use case for this is?

**Comment by johngray1965** [comment]
Sure.  If your setting up the recyclerview to preload items (override getExtraLayoutSpace on LinearLayoutManager), so you might have it load a screen worth of items before and after the visible data, then headers get messed (its picks a row for the header, that's not actually visible).  My change fixes the calculations for figuring out which row the header will be based on is to make sure its actually a visible row.

File: library/src/main/java/com/timehop/stickyheadersrecyclerview/StickyRecyclerHeadersVisibleAdapter.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package com.timehop.stickyheadersrecyclerview;
+
+/**
+ * Created by gray on 2/12/16.
+ */
```
Comment (jacobtabak) [inline library/src/main/java/com/timehop/stickyheadersrecyclerview/StickyRecyclerHeadersVisibleAdapter.java]: Please remove `created by` header

File: library/src/main/java/com/timehop/stickyheadersrecyclerview/StickyRecyclerHeadersVisibleAdapter.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+public interface StickyRecyclerHeadersVisibleAdapter {
```
Comment (jacobtabak) [inline library/src/main/java/com/timehop/stickyheadersrecyclerview/StickyRecyclerHeadersVisibleAdapter.java]: Not liking the name of this class... maybe just `ItemVisibilityAdapter`?

Comment (jacobtabak) [inline library/src/main/java/com/timehop/stickyheadersrecyclerview/StickyRecyclerHeadersVisibleAdapter.java]: Need javadocs explaining what this is and how it is used

**Comment by jacobtabak** [comment]
We will also need to update the documentation (readme.md) and ideally include this functionality in the sample.

**Comment by johngray1965** [comment]
Ok, I'll work on all that later today or over the weekend.

**Comment by johngray1965** [comment]
I addressed all the concerns except for adding it to the sample app.  Its going to complicate the sample app with a fair bit of things to won't concern most people.  Are you sure you want to do that?

**Comment by jacobtabak** [comment]
No, not sure :)

**Pull request merged**